src/screens/ConversationList/ConversationList.js

```diff
@@ -237,23 +244,40 @@ class ConversationListComponent extends Component {
   };
 
   renderEmptyMessage = () => {
-    return <Empty image={images.emptyConversations} title={i18n.t('CONVERSATION.EMPTY')} />;
+    const {
+      eva: { style },
+    } = this.props;
+    return (
+      <Layout style={style.tabContainer}>
+        <Empty image={images.emptyConversations} title={i18n.t('CONVERSATION.EMPTY')} />
+      </Layout>
+    );
+  };
+
+  onRefresh = () => {
+    this.setState({ refreshing: true });
+    this.loadConversations();
```

Comment:
@muhsin-k I'm not that familiar with react native, so this might be a dumb question. Can we wait (async-await) on `this.loadConversations()` to change back the `refreshing` state to `false`? Or can we return a promise from `loadConversations` function to do a promise chain to change back the `refreshing` state?

Because here we are waiting for just 1 second to change back the `refreshing` state to `false` by which time there are chances that API call might not have completed?

Let me know your thoughts.

Reply:
@sony-mathew You are right. But here the implementation is different.  Whenever we call the method `loadConversations()`, the default loader component  `<ConversationItemLoader/>`(Skeleton loader component) will start rendering.  If we are doing pull to refresh and wait for the complete response, there will be two loaders on the screen. I feel like that is not a good UX. That's why I am just showing pull to refresh the loader for only 1 second.

Reply:
I trust your judgement. 😉